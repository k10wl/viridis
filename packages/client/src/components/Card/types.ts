export type CardT = {
  views: number;
  favorite: boolean;
  imageSrc: string;
  name: string;
  description: string;
  tags: string[];
  onClick: () => void;
};
