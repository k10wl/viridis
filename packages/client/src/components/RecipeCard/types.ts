import { RecipeBase } from '@viridis/shared/types';

export type RecipeProps = Omit<RecipeBase, 'cookingProcess'>;
