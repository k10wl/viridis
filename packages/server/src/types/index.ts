export enum Category {
  coffee = 'Coffee',
  tea = 'Tea',
  snacks = 'Chocolate, Fruits & Nuts',
  accessories = 'Accessories',
}

export enum CoffeeCategories {
  espresso = 'Espresso',
  latte = 'Latte',
  mocha = 'Mocha',
  americano = 'Americano',
  cappuccino = 'Cappuccino',
  breve = 'Breve',
}

export enum TeaCategories {
  black = 'Black',
  green = 'Green',
  dragon = 'Dragon',
  melon = 'Melon',
  jasmine = 'Jasmine',
  sweet = 'Sweet',
}

export enum SnacksCategories {
  chocolate = 'Chocolate',
  fruits = 'Fruits',
  nuts = 'Nuts',
}

export enum AccessoriesCategories {
  grinder = 'Grinder',
  cup = 'Cup',
  jar = 'Jar',
  coffeeMaker = 'Coffee maker',
}

type CoffeeProduct = {
  category: Category.coffee;
  subcategory: CoffeeCategories;
};

type TeaProduct = {
  category: Category.tea;
  subcategory: TeaCategories;
};

type SnackProduct = {
  category: Category.snacks;
  subcategory: SnacksCategories;
};

type AccessoriesProduct = {
  category: Category.accessories;
  subcategory: AccessoriesCategories;
};

export type ProductCategory =
  | CoffeeProduct
  | TeaProduct
  | SnackProduct
  | AccessoriesProduct;

export type ProductType = ProductCategory & {
  _id?: string;
  specification: ProductCategory;
  name: string;
  description?: string;
  tags: string[];
};
