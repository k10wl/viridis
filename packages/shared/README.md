# Package info:

`Shared` package contains and exports constants and types for `client` and `server` packages.

| How to access types and constants from anywhere:          |
|-----------------------------------------------------------|
| `import { TypeName } from @viridis/shared/types`          |
| `import { CONSTANT_NAME } from @viridis/shared/constants` |
