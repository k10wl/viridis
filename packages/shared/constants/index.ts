export * from "./categoriesTree";
export * from "./ingredients";
