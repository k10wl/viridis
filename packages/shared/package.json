{
  "name": "@viridis/shared",
  "version": "0.0.1",
  "license": "MIT",
  "devDependencies": {
    "@typescript-eslint/eslint-plugin": "^5.28.0",
    "@typescript-eslint/parser": "^5.28.0",
    "eslint": "^8.17.0",
    "eslint-config-airbnb-base": "^15.0.0",
    "eslint-config-prettier": "^8.5.0",
    "eslint-import-resolver-typescript": "^2.7.1",
    "eslint-plugin-import": "^2.26.0",
    "eslint-plugin-prettier": "^4.0.0",
    "prettier": "^2.7.1",
    "prettier-eslint": "^15.0.1",
    "ts-node": "^10.8.1",
    "typescript": "^4.7.3"
  }
}
