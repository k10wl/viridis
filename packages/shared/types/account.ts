export enum PERMISSION_GROUP {
  USER = "user",
  ADMIN = "admin",
  MAINTAINER = "maintainer",
}

export interface AccountT {
  _id: string;
  name: string;
  password: string;
  permission: keyof typeof PERMISSION_GROUP;
  likedRecipes: string[];
}
