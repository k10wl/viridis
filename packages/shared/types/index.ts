export { PERMISSION_GROUP, AccountT } from "./account";
export * from "./recipe";
export * from "./categories";
