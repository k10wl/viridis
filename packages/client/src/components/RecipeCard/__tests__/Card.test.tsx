import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import reactRenderer from 'react-test-renderer';

import { Ingredient } from '@viridis/shared/types';

import { RecipeCard } from '../index';

jest.mock('react-router', () => ({
  useNavigate: (): (() => void) => jest.fn(),
}));

const INGREDIENTS: Ingredient[] = [
  {
    name: 'авокадо',
    amount: {
      type: 'teaspoon',
      value: 'teaspoon',
    },
  },
  {
    name: 'манго',
    amount: {
      type: 'teaspoon',
      value: 'teaspoon',
    },
  },
];

const PROPERTIES = {
  picture: 'Eheu, homo!',
  name: 'Orexis, orgia, et ausus.',
  description: 'Cur turpis observare?',
  ingredients: INGREDIENTS,
};

test('RecipeCard component', () => {
  const tree = reactRenderer
    .create(
      <RecipeCard
        id="1"
        picture={PROPERTIES.picture}
        name={PROPERTIES.name}
        description={PROPERTIES.description}
        ingredients={PROPERTIES.ingredients}
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();

  render(
    <RecipeCard
      id="1"
      picture={PROPERTIES.picture}
      name={PROPERTIES.name}
      description={PROPERTIES.description}
      ingredients={PROPERTIES.ingredients}
    />,
  );

  const cardNode = screen.getByTestId('card');
  const nameNode = screen.getByText(PROPERTIES.name);
  const imageNode = screen.getByTestId('image');
  const descriptionNode = screen.getByText(PROPERTIES.description);

  expect(cardNode).toBeInTheDocument();
  expect(nameNode).toBeInTheDocument();
  expect(imageNode).toBeInTheDocument();
  expect(imageNode).toHaveAttribute('src', PROPERTIES.picture);
  expect(descriptionNode).toBeInTheDocument();

  PROPERTIES.ingredients.forEach(ingredient => {
    expect(screen.queryByText(ingredient.name)).toBeInTheDocument();
  });

  fireEvent.click(cardNode);

  fireEvent.keyPress(cardNode, {
    key: 'Enter',
    code: 'Enter',
    charCode: 13,
  });
});
