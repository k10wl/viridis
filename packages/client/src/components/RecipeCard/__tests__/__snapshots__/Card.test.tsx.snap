// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard component 1`] = `
.c10 {
  min-width: 100%;
  height: 30px;
  outline: none;
  border: none;
  border-radius: 4px;
  font-size: 1rem;
  font-family: 'Roboto',sans-serif;
}

.c10:hover {
  cursor: pointer;
}

.c10:focus-visible {
  outline: 3px solid #C6725880;
  outline-offset: 1px;
}

.c1 {
  margin: 0;
}

.c2 {
  min-height: 255px;
}

.c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  height: 255px;
  min-height: 255px;
  width: 100%;
  object-fit: cover;
}

.c4 {
  padding: 0 10px;
  margin: 5px 0 0;
  color: #121111;
  font-family: 'PT Serif Bold',serif;
  font-size: 1.25rem;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c5 {
  padding: 0 10px;
  margin: 7px 0 6px;
  color: #696262;
  font-family: 'Roboto Thin',sans-serif;
  font-size: 0.875rem;
  text-overflow: ellipsis;
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  white-space: normal;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  height: 100%;
  overflow: hidden;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 5px;
  padding: 5px 10px;
  margin: 0;
  max-height: 45px;
  list-style-position: inside;
  overflow: hidden;
}

.c8 {
  padding: 3px 10px 3px 4px;
  border-radius: 4px;
  text-align: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #FBFDFD;
  background-color: #244E4B;
  font-family: 'Roboto Thin',sans-serif;
  font-size: 0.75rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 300px;
  max-width: 300px;
  height: 445px;
  max-height: 445px;
  border: 1px solid black;
  border-radius: 7px;
  overflow: hidden;
  cursor: pointer;
  background: rgba(36,78,75,0);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9 {
  padding: 5px;
}

<div
  className="c0"
  data-testid="card"
  onClick={[Function]}
  onKeyPress={[Function]}
  style={
    Object {
      "backgroundColor": "#FBFDFD",
    }
  }
  tabIndex={0}
>
  <figure
    className="c1"
  >
    <div
      className="c2"
    >
      <img
        alt="Orexis, orgia, et ausus."
        className="c3"
        data-testid="image"
        src="Eheu, homo!"
      />
    </div>
    <figcaption
      className="c4"
    >
      Orexis, orgia, et ausus.
    </figcaption>
    <p
      className="c5"
    >
      Cur turpis observare?
    </p>
  </figure>
  <div
    className="c6"
  >
    <ul
      className="c7"
    >
      <li
        className="c8"
      >
        авокадо
      </li>
      <li
        className="c8"
      >
        манго
      </li>
    </ul>
  </div>
  <div
    className="c9"
  >
    <button
      className="c10"
      height="30px"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#C67258",
          "border": "#C67258 solid 0",
          "color": "#FBFDFD",
        }
      }
      type="button"
      width="100%"
    >
      Open recipe in new page
    </button>
  </div>
</div>
`;
